diff --git a/src/util.hpp b/src/util.hpp
index 0d6c379..62702d5 100644
--- a/src/util.hpp
+++ b/src/util.hpp
@@ -35,11 +35,12 @@ template <typename Int, typename Int2>
 constexpr inline Int cdiv(Int a, Int2 b) { return (a + b - 1) / b; }
 
 #ifdef _WIN32
-#define NOMINMAX
 #include <windows.h>
 #include <processthreadsapi.h>
-#include "uint128_t.h"
-#else
+#ifndef _popcnt64
+#define __popcnt64 _popcnt64
+#endif
+#endif
 // __uint__128_t is only available in 64 bit architectures and on certain
 // compilers.
 typedef __uint128_t uint128_t;
@@ -52,8 +53,6 @@ std::ostream &operator<<(std::ostream &strm, uint128_t const &v)
     return strm;
 }
 
-#endif
-
 // compiler-specific byte swap macros.
 #if defined(_MSC_VER)
 
